app/code/community/Razorpay/Payments/Helper/Data.php

```diff
@@ -1,25 +1,148 @@
+            'refund'    => self::BASE_URL . 'payments/:id/refund'
+        );
+
+        $this->userAgent = Mage::getModel(self::PAYMENT_MODEL)->_getChannel();
+    }
+
+    public function isRazorpayEnabled()
+    {
+        return Mage::getStoreConfigFlag(self::CONFIG_PATH_RAZORPAY_ENABLED);
+    }
+
+    public function createOrder($receipt, $amount)
+    {
+        if ($this->isRazorpayEnabled())
+        {
+            $currency = Razorpay_Payments_Model_Paymentmethod::CURRENCY;
+
+            $url = $this->getRelativeUrl('order');
+
+            $postData = array(
+                'receipt'   => $receipt,
+                'amount'    => $amount,
+                'currency'  => $currency
+            );
+
+            $response = $this->sendRequest($url, $postData);
+
+            $returnArray = array(
+                'razorpay_order_id'  => $response['id']
+            );
+
+            return $returnArray;
+        }
+
+        throw new Exception('MAGENTO_ERROR: Payment Method not available');
+    }
+
+    public function capturePayment($paymentId, $amount)
+    {
+        if ($this->isRazorpayEnabled())
+        {
+            $url = $this->getRelativeUrl('capture', array(
+                ':id'   => $paymentId
+            ));
+
+            $postData = array(
+                'amount'    => $amount
+            );
+
+            $response = $this->sendRequest($url, $postData);
+
+            if ($response['status'] === 'captured')
+            {
+                return true;
+            }
+
+            throw new Exception('Capture_Error: Unable to capture payment ' . $paymentId);
+        }
+
+        throw new Exception('MAGENTO_ERROR: Payment Method not available');
+    }
+
+    public function sendRequest($url, $content, $method = 'POST')
+    {
+        $paymentModel = Mage::getModel(self::PAYMENT_MODEL);
+
+        $keyId = $paymentModel->getConfigData(self::KEY_ID);
+        $keySecret = $paymentModel->getConfigData(self::KEY_SECRET);
+
+        $ch = curl_init();
+
+        curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_USERPWD, $keyId . ":" . $keySecret);
+        curl_setopt($ch, CURLOPT_TIMEOUT, 60);
+        curl_setopt($ch, CURLOPT_USERAGENT, $this->userAgent);
+
+        if ($method === 'POST')
+        {
+            curl_setopt($ch, CURLOPT_POST, 1);
+            curl_setopt($ch, CURLOPT_POSTFIELDS, $content);
+        }
+
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+
+        $response = curl_exec($ch);
+        $httpStatus = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+
+        curl_close($ch);
+
+        if ($response === false)
+        {
+            $error = 'CURL_ERROR: ' . curl_error($ch);
+
+            throw new Exception($error);
+        }
+        else
+        {
+            $responseArray = json_decode($response, true);
+
+            if ($httpStatus === 200 and isset($responseArray['error']) === false)
```

Comment:
Not necessarily. For capture, yes, but for some other routes we might return 204 or similar